### PR TITLE
tests: add fuse interface spread test

### PR DESCRIPTION
**Pull request by fgimenez**
Hi @stgraber :) this is the test I mentioned. It uses the same structure of the interface spread tests that we are using so far, you can take a look at https://github.com/snapcore/snapd/tree/master/tests/main, all the tasks prefixed with `interfaces-`.

Usually we create the snaps with the snapbuild tool, built from source https://github.com/snapcore/snapd/tree/master/tests/lib/snapbuild, but for this test this shouldn't work, it is not able to add parts to a snap and in this case we require dependencies for being able to create a fuse filesystem. Maybe there's a better way of doing this than using python and pyhton-fuse, please let me know what do you think.

The test snap uses the hello.py example file that comes with python-fuse http://packages.ubuntu.com/xenial/amd64/python-fuse/filelist, it creates a fuse filesystem in the mount point given as parameter and a `hello` file in there with the `Hello World!` content. I've tried it locally, with the python-fuse package installed and it works fine.

However, the snap with the plug connected is failing, executing fuse-consumer.create with a path writable for the snap (under /var/snap/fuse-consumer/current for instance) returns `Bad system call`, from dmesg I get:
```
[30231.486673] audit: type=1326 audit(1469806014.829:353): auid=4294967295 uid=1000 gid=1000 ses=4294967295 pid=28553 comm="python" exe="/snap/fuse-consumer/x1/usr/bin/python2.7" sig=31 arch=c000003e syscall=165 compat=0 ip=0x7f42b3ab0e2a code=0x0
```
syscall 165 seems to be `mount` and the entr point `sys_mount`.

Please take a look at the `tests/main/interfaces-fuse/task.yaml` file in this changeset to see the exact steps to reproduce, let me know if something isn't clear.

Cheers!

### CODE REVIEW

**Comment by fgimenez** [comment]
@stgraber Quick question, are the snaps declaring a plug on this interface intended to be able to create the file system or only using a fs created elsewhere? The snap in this test tries to create the fs and then make assertions about it from the outside, please let me know if that's the intended usage of the interface. 

I've also tried to access a file system created externally with a simpler snap that declares the fuse plug and which command just cats a file and that works on paths readable by the snap no matter the status of the plug (connected or disconnected), in other non-readable paths it fails (also independently of the status of the plug).

Thanks :)

**Comment by stgraber** [comment]
So because of the way the app launcher works, the mount generated by a fuse filesystem won't be visible to the outside. That's why I mentioned adding the ability to share the mount namespace between binaries of the same snap in https://bugs.launchpad.net/snappy/+bug/1606510.

Thanks for your work! I'm off today but will merge your test into my branch and test/tweak this tomorrow so we can hopefully get it merged soon.

File: tests/main/interfaces-fuse/task.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+summary: Ensure that the fuse interface works.
+
+details: |
+    The fuse interfac allows a snap to manage FUSE file systems.
```
Comment (elopio) [inline tests/main/interfaces-fuse/task.yaml]: typo: interface.

Comment (fgimenez) [inline tests/main/interfaces-fuse/task.yaml]: @elopio fixed, thx!

**Pull request merged**